src/td.hh

```diff
@@ -0,0 +1,247 @@
+#pragma once
+
+#include <memory>
+
+#include "common/panic.hh"
+#include "container/task.hh"
+#include "future.hh"
+#include "scheduler.hh"
+#include "sync.hh"
+
+namespace td
+{
+// Divide ints and round up
+// a > 0, b > 0
+template <class T = int>
+constexpr T int_div_ceil(T a, T b)
+{
+    return 1 + ((a - 1) / b);
+}
+
+// == wait ==
+
+inline void wait_for(sync& sync) { Scheduler::current().wait(sync, true); }
+inline void wait_for_unpinned(sync& sync) { Scheduler::current().wait(sync); }
+
+template <class... STs>
+void wait_for(sync& s, sync& peek, STs&... tail)
+{
+    wait_for(s);
+    wait_for(peek, tail...);
+}
+
+template <class... STs>
+void wait_for_unpinned(sync& s, sync& peek, STs&... tail)
+{
+    wait_for_unpinned(s);
+    wait_for_unpinned(peek, tail...);
+}

```

Comment:
this doesn't need to be recursive, just use a fold expression. much better on the compile times ;)